tracer/src/Datadog.Trace/Telemetry/Collectors/ConfigurationTelemetry.Collector.cs

```diff
@@ -8,14 +8,28 @@
 using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.Globalization;
 using System.Threading;
+using Datadog.Trace.Logging;
 using Datadog.Trace.Telemetry;
 
 namespace Datadog.Trace.Configuration.Telemetry
 {
     // This is the "collector" implementation
     internal partial class ConfigurationTelemetry
     {
+        private static readonly IReadOnlyDictionary<string, string> ConfigurationKeysMapping = new Dictionary<string, string>
+        {
+            ["DD_TRACE_ENABLED"] = "trace_enabled",
+            ["DD_PROFILING_ENABLED"] = "profiling_enabled",
+            ["DD_APPSEC_ENABLED"] = "appsec_enabled",
+            ["DD_DATA_STREAMS_ENABLED"] = "data_streams_enabled",
+            ["DD_TAGS"] = "trace_tags",
```

Comment:
And this is why we were pushing to use the transformed values... Because there are other potential values here, like `DD_TRACE_GLOBAL_TAGS` 😉